drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -1415,7 +1415,14 @@ RDD::TextureID RenderingDeviceDriverD3D12::texture_create(const TextureFormat &p
 	}
 	tex_info->states_ptr = &tex_info->owner_info.states;
 	tex_info->format = p_format.format;
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
 	tex_info->desc = *(CD3DX12_RESOURCE_DESC *)&resource_desc;
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
```

Comment:
Alternatively, the `desc` type could be updated from `CD3DX12_RESOURCE_DESC` to `CD3DX12_RESOURCE_DESC1`; the former is natively convertible to the latter, circumventing the need for pointer shenanigans entirely

Reply:
CC @RandomShaper